nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -618,30 +438,205 @@ in {
+                By default, the following properties are set to “Show to everyone”
+                if this flag is enabled:
+                - About
+                - Full name
+                - Headline
+                - Organisation
+                - Profile picture
+                - Role
+                - Twitter
+                - Website
+                Only has an effect in Nextcloud 23 and later.
+              '';
+            };
+          };
+          objectstore = {
+            s3 = {
+              enable = mkEnableOption (lib.mdDoc ''
+                S3 object storage as primary storage.
+
+                This mounts a bucket on an Amazon S3 object storage or compatible
+                implementation into the virtual filesystem.
+
+                Further details about this feature can be found in the
+                [upstream documentation](https://docs.nextcloud.com/server/22/admin_manual/configuration_files/primary_storage.html).
+              '');
+              bucket = mkOption {
+                type = types.str;
+                example = "nextcloud";
+                description = lib.mdDoc ''
+                  The name of the S3 bucket.
+                '';
+              };
+              autocreate = mkOption {
+                type = types.bool;
+                description = lib.mdDoc ''
+                  Create the objectstore if it does not exist.
+                '';
+              };
+              key = mkOption {
+                type = types.str;
+                example = "EJ39ITYZEUH5BGWDRUFY";
+                description = lib.mdDoc ''
+                  The access key for the S3 bucket.
+                '';
+              };
+              secretFile = mkOption {
+                type = types.str;
+                example = "/var/nextcloud-objectstore-s3-secret";
+                description = lib.mdDoc ''
+                  The full path to a file that contains the access secret. Must be
+                  readable by user `nextcloud`.
+                '';
+              };
+              hostname = mkOption {
+                type = types.nullOr types.str;
+                default = null;
+                example = "example.com";
+                description = lib.mdDoc ''
+                  Required for some non-Amazon implementations.
+                '';
+              };
+              port = mkOption {
+                type = types.nullOr types.port;
+                default = null;
+                description = lib.mdDoc ''
+                  Required for some non-Amazon implementations.
+                '';
+              };
+              useSsl = mkOption {
+                type = types.bool;
+                default = true;
+                description = lib.mdDoc ''
+                  Use SSL for objectstore access.
+                '';
+              };
+              region = mkOption {
+                type = types.nullOr types.str;
+                default = null;
```

Comment:
It's a bummer that this wasn't the case in the past, but now that we're writing this verbatim into Nextcloud's config structure, we should also adhere to their defaults. See https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/primary_storage.html, e.g. region defaults to `eu-west-1`.